android/src/main/java/com/reactnativecommunity/webview/RNCWebChromeClient.java

```diff
@@ -208,7 +209,18 @@ public void onGeolocationPermissionsShowPrompt(String origin, GeolocationPermiss
             requestPermissions(Collections.singletonList(Manifest.permission.ACCESS_FINE_LOCATION));
 
         } else {
-            callback.invoke(origin, true, false);
+            String alertMessage = String.format("Allow this app to use your location?");
+            AlertDialog.Builder builder = new AlertDialog.Builder(this.mWebView.getContext());
+            builder.setMessage(alertMessage);
+            builder.setCancelable(false);
+            builder.setPositiveButton("Allow", (dialog, which) -> {
+                callback.invoke(origin, true, false);
+            });
+            builder.setNegativeButton("Don't allow", (dialog, which) -> {
+                callback.invoke(origin, false, false);
+            });
+            AlertDialog alertDialog = builder.create();
+            alertDialog.show();
```

Comment:
We talked about adding the this.delayed function here on the mobile patch, will this be addressed later ?

Reply:
Nevermind it was updated here on this [PR](https://github.com/MetaMask/react-native-webview-mm/pull/10)